ospfd/ospf_te.c

```diff
@@ -1244,7 +1244,10 @@ ospf_mpls_te_lsa_new (struct ospf_area *area, struct mpls_te_link *lp)
       tmp = SET_OPAQUE_LSID (OPAQUE_TYPE_INTER_AS_LSA, lp->instance);
       lsa_id.s_addr = htonl (tmp);
 
-      struct ospf *top = ospf_lookup ();
+      struct ospf *top = area->ospf;
```

Comment:
Even if struct ospf could be retrieve from area variable, in the case of INTER_AS LSA flood in the AS (LSA Type 11), the area input variable is not defined i.e. could be null. So, it is not safer to get the struct ospf pointer from area->ospf. It is why I use ospf_lookup() call instead. Otherwise, area is defined and could be used.

So, it must be change to:
struct ospf *top = ospf_lookup() or like appear at line 1552 by:

if (area)
   top = area->ospf;
else
    top = ospf_lookup ();
But, this imply to declare struct ospf *top before